lib/src/kbasesearchengine/parse/KeywordParser.java

```diff
@@ -33,6 +34,19 @@ public static ParsedObject extractKeywords(
             final ObjectLookupProvider lookup,
             final List<GUID> objectRefPath)
             throws IOException, ObjectParseException, IndexingException, InterruptedException {
+
+        // check pre-conditons
+        if(json == null) throw new IllegalArgumentException("json is a required parameter");
```

Comment:
Looks like we have some style differences we should resolve for the sake of consistency in the codebase:
* I leave spaces after keywords so they don't look like fn calls (if here and for and catch below)
* I never use bracketless blocks, even when the entire if statement is on one line (goto fail bug-ish)

Reply:
Fixed